### DISCUSSION TITLE
[Feature Request]:SLURP

### DISCUSSION THREAD

**Discussion by zhengqing187** [speechbrain/speechbrain#2006] [Q&A]
### 🚀 The feature

hello

I would like to consult the experiment on SLURP data set. Do you combine the two tasks of intention detection and slot filling? Does the loss function add the loss function of the two tasks?

### Solution outline

learn more about the model

**Comment by BenoitWang** [comment]
Hi @zhengqing187, sorry I just saw this. Our SLU is treated as a seq2seq token generation task. For example we train the model to decode strings like `"{'scenario': 'alarm'| 'action': 'set'| 'entities': [{'type': 'time'| 'filler': 'ten'}]}"`. So it'd be enough to calculate the NLL loss over the decoded & ground-truth tokens.

During evaluation we only need to reformat the decoded string to match the calculation of accuracy.
